dgl_ptm/dgl_ptm/model/step.py

```diff
@@ -75,7 +76,8 @@ def ptm_step(agent_graph, device, timestep, params):
         agent_update(agent_graph, params, device=device, method ='income')
         agent_update(agent_graph, params, device=device, method ='consumption')
 
-        #Data collection and storage
-        data_collection(agent_graph, timestep = timestep, npath = params['npath'], epath = params['epath'], ndata = params['ndata'], 
-                        edata = params['edata'], mode = params['mode'])
+        if (0 < params['data_collection_period'] and timestep % params['data_collection_period'] == 0) or (params['data_collection_list'] and timestep in params['data_collection_list']):
+            #Data collection and storage
+            data_collection(agent_graph, timestep = timestep, npath = params['npath'], epath = params['epath'], ndata = params['ndata'], 
+                            edata = params['edata'], mode = params['mode'])
```

Comment:
```suggestion
# do data collection at every step as an nitegeror at specific time steps as
    # a list of integers defined by `data_collection_intervals`
    do_data_collection = False
    if isinstance(params['data_collection_intervals'], int):
        if timestep%params['data_collection_intervals']==0:
            do_data_collection = True
    else:
        if timestep in params['data_collection_intervals']:
            do_data_collection = True

    if do_data_collection:
        #Data collection and storage
        data_collection(agent_graph, timestep = timestep, npath = params['npath'], epath = params['epath'], ndata = params['ndata'],
                        edata = params['edata'], mode = params['mode'])
```

Reply:
The suggestion makes the if statement more readable and avoids duplicating codes of `data_collection(....)`